server/base/src/main/java/org/apache/accumulo/server/conf/TableConfiguration.java

```diff
@@ -178,6 +178,14 @@ private static CompactionDispatcher createCompactionDispatcher(AccumuloConfigura
     CompactionDispatcher newDispatcher = Property.createTableInstanceFromPropertyName(conf,
         Property.TABLE_COMPACTION_DISPATCHER, CompactionDispatcher.class, null);
 
+    if (newDispatcher == null) {
+      // return early to prevent NPE
+      log.error(
```

Comment:
Does it need to be an error however?  Could it be a debug or info?

Reply:
In the case where an invalid classloader context is set, then `Property.createTableInstanceFromPropertyName` cannot even return the *default* CompactionDispatcher specified in `Property.TABLE_COMPACTION_DISPATCHER`. This being null is a symptom of a misconfiguration of the context class loader. I think it's correct to point out the error here.

Reply:
GIven the changes in #3678 , this message should hopefully not appear that much anymore. I think it's fine to leave this at ERROR in this case because that would mean a failure of the check in #3678